.travis.yml

```diff
@@ -4,20 +4,21 @@ env:
 os:
   - osx
 
-sudo: false
-
 language: go
 
-go:
-  - 1.13.x
-
-script:
-  - df -h
-  - travis_wait make ci
-
 branches:
   only:
     - master
 
-matrix:
+jobs:
   fast_finish: true
+  include:
+    - go: "1.13.x"
+      name: "go test"
+      script:
+        - df -h
+        - travis_wait make ci
+    - go: "1.13.x"
+      name: "go lint"
+      script: travis_wait make ci-lint
```

Comment:
I'd tests this in circle-ci even may be in place of check-fmt: 

https://github.com/hashicorp/packer/blob/44ba58321d9be80fa418e4dfc84c4e588a2bc309/.circleci/config.yml#L81-L85

Reply:
Ah I thought about this as well. But golangci-lint is currently being executed with the `--new-from-rev` flag so it will only check against newly added files. Where `make fmt-check` will validate existing and new so it wouldn't be a fair swap. In the future, if we enable golangci-lint on all files we can swap out for sure.

Re:circle-ci happy to move it there instead of on Travis. Is there a advantage? Or is the idea to add all new ci jobs in circle?

Reply:
Ah right, but shouldn't everything be already fmt'ed already and since `--new-from-rev` is ran then the new impacted stuff will be fmt'ed ? I think the outcome will be the same 🤔.

```
  -n, --new                            Show only new issues: if there are unstaged changes or untracked files, only those changes are analyzed, else only changes in HEAD~ are analyzed.
                                       It's a super-useful option for integration of golangci-lint into existing large codebase.
                                       It's not practical to fix all existing issues at the moment of integration: much better to not allow issues in new code.
                                       For CI setups, prefer --new-from-rev=HEAD~, as --new can skip linting the current patch if any scripts generate unstaged files before golangci-lint runs.
      --new-from-rev REV               Show only new issues created after git revision REV
      --new-from-patch PATH            Show only new issues created in git patch with file path PATH
```

The only advantage I can think of is that travis is much more unstable and subject to lags/downs than circle, from my experience.

Reply:
> Ah right, but shouldn't everything be already fmt'ed already and since --new-from-rev is ran then the new impacted stuff will be fmt'ed ? I think the outcome will be the same 

As of now everything is fmt'ed but if an existing file is modified and not fmt'ed locally before creating the PR golangci-lint with the `--new-from-rev ` will not find the issue.

The description of the new flag is a bit misleading; by new issues it means issues found in new files, not new issues in existing files. If the existing file `./builder/amazon/ebs/builder.go` is modified and the new change is not formatted (e.g goimports is out of order), running golangci-lint will not analyze that file because it is not a new file. I originally had a check that would run golangci-lint against modified/added files only but that was showing all issues found and not the ones introduced by the commits. I think we can revisit that option once we've fixed the existing linting issues.

Reply:
Ah, yes, I thought `Show only new issues` meant `Show only new issues` ! Lol. Okay gotcha ! 👍

Reply:
Yeah same here.. I spent some time looking at it only to realize they really mean new :laughing: The good thing is the `--new-from-patch` will give us what we want in the future :crossed_fingers: